sycl/source/detail/global_handler.hpp

```diff
@@ -0,0 +1,76 @@
+//==--------- global_handler.hpp --- Global objects handler ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl/detail/spinlock.hpp>
+#include <CL/sycl/detail/util.hpp>
+#include <detail/platform_impl.hpp>
+#include <detail/plugin.hpp>
+#include <detail/program_manager/program_manager.hpp>
+#include <detail/scheduler/scheduler.hpp>
+
+#include <memory>
+#include <vector>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace detail {
+using PlatformImplPtr = std::shared_ptr<platform_impl>;
+
+struct GlobalHandler;
```

Comment:
Please, add a comment clarifying why `GlobalHandler` class is forward declared as a struct right before the class declaration.

Reply:
Leftovers from early prototypes. Removed.